daemon/list_test.go

```diff
@@ -79,7 +81,46 @@ func containerListContainsName(containers []*containertypes.Summary, name string
 	return false
 }
 
-func TestListInvalidFilter(t *testing.T) {
+func TestContainerList(t *testing.T) {
+	db, err := container.NewViewDB()
+	assert.NilError(t, err)
+	d := &Daemon{
+		containersReplica: db,
+	}
+
+	// test list with different number of containers
+	for _, num := range []int{0, 1, 2, 4, 8, 16, 32, 64, 100} {
+		t.Run(fmt.Sprintf("%d containers", num), func(t *testing.T) {
+			db, err := container.NewViewDB() // new DB to ignore prior containers
+			assert.NilError(t, err)
+			d = &Daemon{
+				containersReplica: db,
+			}
+
+			// create the containers
+			containers := make([]*container.Container, num)
+			for i := range num {
+				name := fmt.Sprintf("cont-%d", i)
+				containers[i] = setupContainerWithName(t, name, d)
+				// ensure container timestamps are separated enough so the
+				// sort used by d.Containers() can deterministically sort them.
+				containers[i].Created.Add(time.Millisecond)
```

Comment:
Ah, we also need to re-assign the value back because `time.Add` returns a new value rather than mutating the input:

```suggestion
				containers[i].Created = containers[i].Created.Add(time.Millisecond)
```

Reply:
Also, now that we don't delay the test, we can even increment it by larger time intervals, like `time.Second`.

Reply:
Good catch, thanks! Fixed.